ioctl/util/util.go

```diff
@@ -64,6 +77,44 @@ func ConnectToEndpoint(secure bool) (*grpc.ClientConn, error) {
 	return grpc.Dial(endpoint, grpc.WithTransportCredentials(credentials.NewTLS(&tls.Config{})))
 }
 
+// GetGPRCConnection return a shared gPRC connection
+func GetGPRCConnection() (*grpc.ClientConn, error) {
+	gPRCMu.Lock()
+	defer gPRCMu.Unlock()
+
+	if gPRCConnInstance != nil {
+		gPRCTimer.Reset(gRPCConnectionTimeOut)
+		return gPRCConnInstance, nil
+	}
+
+	conn, err := ConnectToEndpoint(config.ReadConfig.SecureConnect && !config.Insecure)
+	if err != nil {
+		return nil, output.NewError(output.NetworkError, "failed to connect to endpoint", err)
+	}
+	gPRCConnInstance = conn
+	//Automatically close gRPC connection after predefined time
+	gPRCTimer = time.AfterFunc(gRPCConnectionTimeOut, func() {
+		conn.Close()
+		gPRCConnInstance = nil
```

Comment:
this runs in another go-routine, so it could still have problem:
at timeOut - 200ms, `GetGPRCConnection` is called, and current connection is returned
after 200ms, (while user is still using it), timeOut fires, and closed the connection

so better to have a `lastActiveUseTs` to keep the last active use timestamp, and if `lastActiveUseTs + timeOut < time.Now`, close the connection